38_EXRSplit/main.cpp

```diff
@@ -1,94 +1,89 @@
-		imgViewParams.subresourceRange = { static_cast<IImage::E_ASPECT_FLAGS>(0u),0u,1u,0u,1u };
-		auto imageView = ICPUImageView::create(std::move(imgViewParams));
+		auto assetManager = make_smart_refctd_ptr<nbl::asset::IAssetManager>(smart_refctd_ptr(m_system));
 
-		auto channelsName = metadata->m_name;
-
-		std::filesystem::path filename, extension;
-		core::splitFilename(filePath.c_str(), nullptr, &filename, &extension);
-		const std::string finalFileNameWithExtension = filename.string() + extension.string();
-		const std::string finalOutputPath = channelsName.empty() ? (filename.string() + "_" + std::to_string(i++) + extension.string()) : (filename.string() + "_" + channelsName + extension.string());
+		// when assets are retrieved you need to provide parameters that control the loading process
+		nbl::asset::IAssetLoader::SAssetLoadParams lp;
+		// at the very least you need to provide the `workingDirectory` if your asset depends on others
+		// this helps resolve relative paths for things such as textures
+		lp.workingDirectory = mediaWD;
+		
 
-		const auto writeParams = IAssetWriter::SAssetWriteParams(imageView.get(), EWF_BINARY);
+		auto checkedLoad = [&]<class T>(const string & key)->smart_refctd_ptr<T>
 		{
-			bool status = assetManager->writeAsset(finalOutputPath, writeParams);
-			assert(status);
+			nbl::asset::SAssetBundle bundle = assetManager->getAsset(key, lp);
+			if (bundle.getContents().empty())
+			{
+				m_logger->log("Asset %s failed to load! Are you sure it exists?", ILogger::ELL_ERROR, key.c_str());
+				return nullptr;
+			}
+			// All assets derive from `nbl::asset::IAsset`, and can be casted down if the type matches
+			static_assert(std::is_base_of_v<nbl::asset::IAsset, T>);
+			// The type of the root assets in the bundle is not known until runtime, so this is kinda like a `dynamic_cast` which will return nullptr on type mismatch
+			auto typedAsset = IAsset::castDown<T>(bundle.getContents()[0]); // just grab the first asset in the bundle
+			if (!typedAsset)
+				m_logger->log("Asset type mismatch want %d got %d !", ILogger::ELL_ERROR, T::AssetType, bundle.getAssetType());
+			return typedAsset;
+		};
+
+		if (auto cpuImage = checkedLoad.operator() < nbl::asset::ICPUImage > ("noises/spp_benchmark_4k_512.exr"))
+		{
+			ICPUImageView::SCreationParams imgViewParams;
+			imgViewParams.flags = static_cast<ICPUImageView::E_CREATE_FLAGS>(0u);
+			imgViewParams.format = cpuImage->getCreationParameters().format;
+			imgViewParams.image = core::smart_refctd_ptr<ICPUImage>(cpuImage);
+			imgViewParams.viewType = ICPUImageView::ET_2D;
+			imgViewParams.subresourceRange = { static_cast<IImage::E_ASPECT_FLAGS>(0u),0u,1u,0u,1u };
+			smart_refctd_ptr<nbl::asset::ICPUImageView> imageView = ICPUImageView::create(std::move(imgViewParams));
+
+			nbl::asset::IAssetWriter::SAssetWriteParams wp(imageView.get());
+			wp.workingDirectory = CWD;
+			assetManager->writeAsset("sample_out.exr", wp);
```

Comment:
please restore the old behaviour, where each channel is saved as a separate EXR with a filename `_channelName` suffix (or `_i` in case of no name channel)